firebase-components/src/main/java/com/google/firebase/components/ComponentRuntime.java

```diff
@@ -122,16 +125,41 @@ private void discoverComponents(List<Component<?>> componentsToAdd) {
           Log.w(ComponentDiscovery.TAG, "Invalid component registrar.", ex);
         }
       }
+      for (int i = 0; i < processedComponents.size(); i++) {
+        Object[] interfaces = processedComponents.get(i).getProvidedInterfaces().toArray();
+        for (Object anInterface : interfaces) {
+          if (anInterface.toString().contains("kotlinx.coroutines.CoroutineDispatcher")) {
```

Comment:
Since this code is very specific to `CoroutineDispatcher` I think you can add a comment explaining why this needs to happen and use var names specific too, like `existingCoroutineDispatcherInterfaces`. This makes it clear that this handles an special case